File: src/main/java/org/jcodec/containers/mp4/boxes/DataInfoBox.java
package org.jcodec.containers.mp4.boxes;

import org.jcodec.containers.mp4.BoxUtil;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * Creates MP4 file out of a set of samples
 * 
 * @author The JCodec project
 *
 */
public class DataInfoBox extends NodeBox {

    public static String fourcc() {
        return "dinf";
    }

    public static DataInfoBox createDataInfoBox() {
        return new DataInfoBox(new Header(fourcc()));
    }

    public DataInfoBox(Header atom) {
        super(atom);
    }

    public DataRefBox getDref() {
        return BoxUtil.findFirst(this, DataRefBox.class, "dref");
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/DataInfoBox.kt
package org.jcodec.containers.mp4.boxes

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * Creates MP4 file out of a set of samples
 *
 * @author The JCodec project
 */
class DataInfoBox(atom: Header) : NodeBox(atom) {
    val dref: DataRefBox
        get() = findFirst(this, "dref") as DataRefBox

    companion object {
        @JvmStatic
        fun fourcc(): String {
            return "dinf"
        }

        @JvmStatic
        fun createDataInfoBox(): DataInfoBox {
            return DataInfoBox(Header(fourcc()))
        }
    }
}